### PR TITLE
Fix links in the referece

### LINKED ISSUE #24999: Reference links need fixing
see http://www.reddit.com/r/rust/comments/34fq0k/is_there_a_good_list_of_rusts_keywords/cqu7w56?context=3

### PR DESCRIPTION
**Pull request by steveklabnik**
and remove references to the unit type, which no longer exists

Fixes #24999

### CODE REVIEW

File: src/doc/reference.md [hunk 1]
```diff
@@ -653,7 +657,7 @@ There are several kinds of item:
 * [`use` declarations](#use-declarations)
 * [modules](#modules)
 * [functions](#functions)
-* [type definitions](#type-definitions)
+* [type definitions](grammar.html#type-definitions)
```
Comment (bluss) [inline src/doc/reference.md]: I think this should just link to the type aliases section in the same chapter 6.1. I touched this in my PR too (added to this particular list, missing constants).

Comment (steveklabnik) [inline src/doc/reference.md]: I thought this was what the original link was to, maybe i'm wrong

**Comment by steveklabnik** [comment]
Updated!

**Pull request merged**